Marlin/Marlin_main.cpp

```diff
@@ -4381,6 +4557,32 @@ inline void gcode_M226() {
 
 #endif // CHDK || PHOTOGRAPH_PIN
 
+/*
+* M241 - Dwell for a given amount of time in milliseconds (500 by default)
+*/
+void gcode_M241(long num_milliseconds) {
```

Comment:
Could you not use a `uint16_t` for `num_milliseconds` here? That would give you over a minute of possible dwell time. An `unsigned long` gives you over 1000 hours of possible dwell time.